### ISSUE TITLE
Option to deploy node-group workers with instance metadata access blocked

### ISSUE THREAD

**Issue by whereisaaron** [weaveworks/eksctl#1310]
**Why do you want this feature?**

EKS has introduced [native support for assigning IAM Roles to Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/restrict-ec2-credential-access.html) (projecting a token that can be used by Pods against AWS OIDC providers).

This means it is no longer necessary to run `kube2iam` or `kiam` to assign per-Pod IAM Roles. However without one of those services running, containers are not blocked from accessing the instance metadata and credentials. And AWS EKS does not block access for you, instead [recommending to run on each node](https://docs.aws.amazon.com/eks/latest/userguide/restrict-ec2-credential-access.html):

`eksctl` has [support for deploying this feature](https://eksctl.io/usage/iamserviceaccounts/) but it doesn’t indicate whether it also secures then instance endpoint.

```
yum install -y iptables-services
iptables --insert FORWARD 1 --in-interface eni+ --destination 169.254.169.254/32 --jump DROP
iptables-save | tee /etc/sysconfig/iptables 
systemctl enable --now iptables
```

**What feature/behavior/change do you want?**

I would like a per-node-group option to bootstrap nodes with a network block on access to instance metadata, similar to the AWS example above. For me this configuration would be the default, now that IAM Roles have native support for EKS containers.

**Comment by whereisaaron** [comment]
`eksctl` now has built-in support for `iamserviceaccounts` now which is great, but is a partial solution while all pods have access to EC2 instance metadata credentials.

When we used to install `kube2iam` or `kiam`, as well as delegating roles, they would block access to instance metadata credentials. Now we are not installing those, and using `iamserviceaccounts` instead, the second-half of the task needs to be covered too. I would like to see a option like `iam.instanceMetadataAccess` which handles this cleanly.

```
iam:
  withOIDC: true
  instanceMetadataAccess: false
  serviceAccounts:
      ...
```

As a workaround, I am using `preBootstrapCommands` with the [AWS suggested commands](https://docs.aws.amazon.com/en_pv/eks/latest/userguide/restrict-ec2-credential-access.html):

    preBootstrapCommands:
      - yum install -y iptables-services
      - iptables --insert FORWARD 1 --in-interface eni+ --destination 169.254.169.254/32 --jump DROP
      - iptables-save | tee /etc/sysconfig/iptables
      - systemctl enable --now iptables

This is working for me. Though this drops traffic rather than rejecting it, which causes access attempts to stall rather than be rejected. Does anyone know if there is a reason AWS chose the DROP approach?

**Comment by scottyhq** [comment]
Thanks for detailing this work-around @whereisaaron. I wanted to link to some documentation on why this is important for cluster security - https://github.com/freach/kubernetes-security-best-practice/blob/master/AWS.md . 

And also link to a broader issue I opened up to explain security best-practices on clusters https://github.com/weaveworks/eksctl/issues/1460. Feedback welcome there!

**Comment by martina-if** [comment]
I believe this was addressed in https://github.com/weaveworks/eksctl/pull/2625. Please feel free to re-open this issue if you disagree or think there's anything missing.

closed via #2625
